beacon-chain/state/getters.go

```diff
@@ -268,6 +268,18 @@ func (b *BeaconState) NumofValidators() int {
 	return len(b.state.Validators)
 }
 
+// ReadFromEveryValidator reads values from every validator and applies it to the provided function.
+// Warning: This method is potentially unsafe, as it exposes the actual validator registry.
```

Comment:
This unsafe method worries me, is there no other way?

Reply:
I am not sure if there is a better way. The alternative is to basically perform a clone of the whole registry. I could remove this, but we will take a performance hit in epoch transitions

Reply:
Is the performance hit noticeable?

Reply:
It is unfortunately, it would be multiple registry clones

Reply:
Not a fan of this read from every func because of side effects. Can we just read state.Validators() in the state transition and pass that down to funcs that need to read it?

Reply:
That has the potential to lead to an invalid state transition like we saw before. It would be risky passing down the same registry slice, to multiple functions that write to it. I have another way, that would be more verbose but safer than this one. I will try to test it out and see if it works well